app/src/main/java/io/apicurio/registry/events/EventsServiceImpl.java

```diff
@@ -36,12 +37,14 @@
 /**
  * @author Fabian Martinez
  */
+@Startup
```

Comment:
Actually, there is no difference between annotating a bean with this and observing the startup event like we're doing (when you annotate the bean with Startup a synthetic observer is created). +1 to the eager initialization, but this annotation is not required.

Reply:
Another possibility is to listen to the storage-ready event instead of waiting for the startup event like we're doing in the [KafkaSql](https://github.com/Apicurio/apicurio-registry/blob/main/storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlRegistryStorage.java#L195) storage since no events are gonna be fired if the storage has not been configured therefore there is no need to configure the sinks. Anyway, up to you, both approaches should work.

Reply:
Tested removing the `Startup` annotation and the mechanism doesn't work somehow ...
It feels a little strange having to rely on `KafkaSQL storage` as it's not directly involved in this race condition.

Reply:
as agreed, moved to `@PostConstruct`